src/parallel/include/timpi/parallel_implementation.h

```
@@ -3028,12 +3031,17 @@ inline void Communicator::allgather(const T & sendval,
   timpi_assert(this->size());
   recv.resize(this->size());
 
+  static const std::size_t approx_total_buffer_size = 1e8;
```

I do not know we can write an integer as 1e8. I thought is was a float type.

---

It is, but C++ will silently convert float types to integral types, and if the compiler can prove that the float value fits in the integer exactly (as it can with a compile-time constant) then even -Wconversion won't complain about it.

---

I assume that there is a motivation for the choice of this number? Just at first glance it seems kind of arbitrary

---

The total message should be less than 1G.

---

when considering today most the supercomputers  will have 2G-4G per core

---

My thinking exactly, @fdkong.  I basically divided 1GB by a typical sizeof(buffertype) of 8 bytes then shrunk it a little further to be safe.

Though to be sure, I still don't like leaving this out of user control.  If you weren't happy with my still-pretty-arbitrary choice then my next thought would have been to move this from a static local into a Communicator member variable, and I'm still kind of considering it regardless.